src/main/java/com/ciaranmckenna/readinglistapp/controller/AuthorController.java

```diff
@@ -1,87 +1,89 @@
-        List<AuthorRecord> authorRecordList = readingListService.findAllAuthors();
+    private final AuthorServiceImpl authorService;
+
+    @GetMapping("/list")
+    public String findByAuthorNameContainingIgnoreCase(@RequestParam(name = "firstName", required = false) String firstName,
+                                                       @RequestParam(name = "lastName", required = false) String lastName,
+                                                       Model model) {
+        List<AuthorRecord> authorRecordList;
+        if (firstName != null && !firstName.isEmpty() || lastName != null && !lastName.isEmpty()) {
+            authorRecordList = authorService.findByAuthorNameContainingIgnoreCase(firstName, lastName);
+        } else {
+            authorRecordList = authorService.findAllAuthors();
+        }
         model.addAttribute("author", authorRecordList);
         return "authors/list-authors";
     }
 
-    @GetMapping("{id}")
+    @GetMapping("/{id}")
     public String findAuthorById(@PathVariable int id, Model model) throws NotFoundException {
-        Author authorById = readingListService.findAuthorById(id);
+        Author authorById = authorService.findAuthorById(id);
         if (authorById != null){
             model.addAttribute("author", authorById);
         }
         return "authors/list-authors";
     }
 
-    @GetMapping("search")
+    @GetMapping("/search")
     public String showFormForSearch(Model model){
         Author author = new Author();
         model.addAttribute("author", author);
         return "authors/author-search";
     }
 
-    @GetMapping("name")
+    @GetMapping("/name")
     public String findAuthorByNameContaining(@RequestParam String firstName, @RequestParam String lastName, Model model){
 
         if (firstName.isEmpty() && lastName.isEmpty()) {
             firstName = ""; // empty String signifies broadest possible search
             lastName = ""; // empty String signifies broadest possible search
 
         }
-        List<AuthorRecord> authorNameContaining = readingListService.findByAuthorNameContainingIgnoreCase(firstName, lastName);
+        List<AuthorRecord> authorNameContaining = authorService.findByAuthorNameContainingIgnoreCase(firstName, lastName);
         model.addAttribute("author", authorNameContaining);
         return "authors/list-authors";
     }
 
-    @GetMapping("registration")
+    @GetMapping("/registration")
     public String showFormForAdd(Model model){
         Author author = new Author();
         model.addAttribute("author", author);
         return "authors/author-form";
     }
 
-    @PostMapping("add")
+    @PostMapping("/add")
     public String addAuthor(@ModelAttribute("author") Author author){
-        readingListService.addAuthor(author);
+        authorService.addAuthor(author);
         return "redirect:/author/list";
     }
 
-    @GetMapping("update/{authorId}")
+    @GetMapping("/update/{authorId}")
```

Comment:
This method is not performing it's described functionality. Updates are not being saved.

Reply:
I dit not touch this part ... I just updated slash!